include/rcpputils/filesystem_helper.hpp

```diff
@@ -251,6 +251,20 @@ class path
   */
   path parent_path() const
   {
+    // Edge case: empty path
+    if (this->empty()) {
+      return path("");
+    }
+
+    // Edge case: if path only consists of one part, then return '.' or '/'
+    //            depending if the path is absolute or not
+    if (1u == path_as_vector_.size()) {
+      if (this->is_absolute()) {
+        return path(std::string(1, kPreferredSeparator));
+      }
+      return path(".");
+    }
```

Comment:
That wouldn't hold on Windows, would it? The parent path of `C:\` is undefined I would say.

Reply:
Good point. 
It seems clearly documented here: https://en.cppreference.com/w/cpp/filesystem/path/parent_path

The parent of "/" is "/". So, I think on Windows we should expect the parent of "C:\\" is "C:\\".

I've added tests to check for this 9a48203 (I think the logic is correct as-is).